### DISCUSSION TITLE
I cant push anyway. Pls help

### DISCUSSION THREAD

**Discussion by yaserdemet** [community/community#22020] [New to GitHub]
<p>Please help me. I cant push my codes to my github. When i try to push this output appears. Authentication failed for “my repo adress “<br>
I have create new github account same problem go on.</p>

**Comment by airtower-luna** [comment]
<p>Please post the full command you used and its output. If you want to keep the repository name private you can replace it with <code>***</code>, but the details are required to help.</p>

**Comment by yaserdemet** [comment, marked as answer]
<p>Thx u so much for your intereset. The problem has been solved. I have not known to click repo section when creating new token.<br>
Btw are u air traffic controller or aviator ? I concluded  due to your name</p>

**Comment by ghost** [comment]
<p>Hi,</p>
<p>can you share more details on how you managed it?</p>
<p>I keep having my access denied when I try to push… I’ve set up the token and still getting an error</p>
<p>I’ve also created a new account and the problem persists</p>
<p>thank you in advance</p>

**Reply by Kwameamerica** [reply]
Can I get aviator predictor??

**Comment by airtower-luna** [comment]
<aside class="quote no-group" data-username="yaserdemet" data-post="3" data-topic="241388">
<div class="title">
<div class="quote-controls"></div>
 yaserdemet:</div>
<blockquote>
<p>Btw are u air traffic controller or aviator ? I concluded due to your name</p>
</blockquote>
</aside>
<p>Interesting guess, but no. Think magic and elements instead. :smile_cat: :wind_chime:</p>
<aside class="quote no-group" data-username="marcoveiobrasil" data-post="4" data-topic="241388">
<div class="title">
<div class="quote-controls"></div>
 marcoveiobrasil:</div>
<blockquote>
<p>I keep having my access denied when I try to push… I’ve set up the token and still getting an error</p>
</blockquote>
</aside>
<p>See my post above: Please post <em>all</em> relevant info to get help.</p>

**Reply by Kwameamerica** [reply]
Can I get aviator predictor??

**Comment by ghost** [comment]
<p>Hi Airtower,</p>
<p>here goes the error:</p>
<p>marcusviniciusdefreitas@marcuss-macbook-air flipflops % git push</p>
<p>Username for --------: <a href="mailto:m------@gmail.com">m------@gmail.com</a></p>
<p>Password for ‘-------<span class="mention">@gmail.com</span>@github.com’:</p>
<p>remote: Permission to marcoveiobrasil/------.git denied to marcoveiobrasil.</p>
<p>fatal: unable to access ‘<a href="https://github.com/marcoveiobrasil/----.git/" rel="noopener nofollow ugc">https://github.com/marcoveiobrasil/----.git/</a>’: The requested URL returned error: 403</p>
<p>it’s just a test repository for a course I’m doing (I’m a beginner); I’ve set up the token on the developers space, my repository is set up in HTTPS,  I’m using Mac and the Git’s 2.32.0 version</p>
<p>thank you for your help</p>

**Reply by Kwameamerica** [reply]
Can I get aviator predictor??

**Comment by airtower-luna** [comment]
<p>You need to enter your actual GitHub username, not your email address. That you already have a PAT is good, make sure it has the <strong>repo</strong> scope, and <strong>workflow</strong> if you want to use Actions.</p>

**Comment by yaserdemet** [comment]
<p>Hi. A little way to make easier pushing , if u dont want to enter your pasaword and username everytime when pushing , run this code git config --global credential.helper store and git config --global core.autocrlf true. When it s comes to your problem, create a new token but when doing this, mark first box which is repository.</p>
<p>1 Nis 2022 Cum, saat 02:14 tarihinde Marcus De Freitas via GitHub Community &lt;<a href="mailto:githubcommunity@discoursemail.com">githubcommunity@discoursemail.com</a>&gt; şunu yazdı:</p>

**Comment by airtower-luna** [comment]
<p>That is dangerous advice. :warning:</p>
<ul>
<li>
<p><code>credential.helper store</code> saves your credentials on disk unencrypted. In some circumstances that’s okay, but not not as general advice without a warning. <code>credential.helper cache</code> is usually good, because it only keeps the credentials in RAM for a limited time. Combine that with a decent password manager for long-term token storage. See also: <a href="https://git-scm.com/book/en/v2/Git-Tools-Credential-Storage" class="inline-onebox">Git - Credential Storage</a>.</p>
</li>
<li>
<p><code>core.autocrlf</code> has nothing to do with authentication, and modifies line endings under certain circumstances. When you want those changes that’s great, but we’ve had people on the forum who’s files got broken that way.</p>
</li>
</ul>

**Comment by craigjford** [comment]
<p>I have been using GitHub since early January.  Now I am encountering the problem below.  Any help would be greatly appreciated</p>
<p>remote: Support for password authentication was removed on August 13, 2021. Please use a personal access token instead.</p>
<p>remote: Please see <a href="https://github.blog/2020-12-15-token-authentication-requirements-for-git-operations/" class="inline-onebox" rel="noopener nofollow ugc">Token authentication requirements for Git operations | The GitHub Blog</a> for more information.</p>
<p>fatal: Authentication failed for ‘<a href="https://github.com/craigjford/.../" rel="noopener nofollow ugc">https://github.com/craigjford/.../</a>’</p>

**Comment by airtower-luna** [comment]
<p>As far as I know that message appears whenever what you entered as the password isn’t a valid token. Maybe your token expired?</p>
<p>If you need more help please share the full command and output, otherwise it’s hard to guess what might be going on.</p>

**Comment by craigjford** [comment]
<p>Thanks airtower!   I am taking a web development course and never had a problem.  I created a token and now everything is working.  Thanks</p>
